src/CoreWCF.Primitives/src/CoreWCF/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -297,8 +297,8 @@ protected override async Task<(Stream, SecurityMessageProperty)> OnAcceptUpgrade
                 _clientCertificate = certificate2;
                 try
                 {
-                    SecurityToken token = new X509SecurityToken(certificate2, false);
-                    ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = _parent.ClientCertificateAuthenticator.ValidateToken(token);
+                    SecurityToken token = new X509SecurityToken(certificate2, true);
+                    ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = _parent.ClientCertificateAuthenticator.ValidateTokenAsync(token).AsTask().GetAwaiter().GetResult();
```

Comment:
Any reason you told this to clone the certificate when creating an X509SecurityToken? We've already cloned it on line 296 when we make our own instance of X509Certificate2 from the passed in certificate.

Reply:
I followed this comment https://github.com/CoreWCF/CoreWCF/pull/419#discussion_r691691284 but i might misunderstood.